src/ddprof_worker.cc

```diff
@@ -168,15 +168,24 @@ DDRes ddprof_pr_sample(DDProfContext *ctx, perf_event_sample *sample, int pos) {
   unsigned long this_ticks_unwind = __rdtsc();
   DDRes res = unwindstate__unwind(us);
 
+  // Usually we want to send the sample_val, but sometimes we need to process
+  // the event to get the desired value
+  PerfWatcher *watcher = &ctx->watchers[pos];
+  uint64_t sample_val = sample->period;
```

Comment:
minor, this could belong to the perf_ringbuffer file ?

Reply:
I've thought about this several times and I don't know if I agree.  Right now, the ringbuffer implementation is "pure" in the sense that it returns to the user exactly what is in the buffer.  If we push down the business logic here (e.g., overwriting the sample value), then we've added logic which doesn't feel to me like a natural part of ringbuffer deserialization.

On the other hand, the worker is already responsible for consolidating a variety of user inputs and application states (such as the sample events).  This area is probably overdue for a refactor.

What are your thoughts?  Why/what did you think could go in the ringbuffer implementation?

Reply:
The suggestion was indeed not great. Ringbuffer should not use watcher knowledge.

We should indeed avoid adding more responsibility to the worker.
I would suggest a simple refactor to have a function computing the value based on watcher + raw memory / registers. 

Minor: Should we not protect the access to the raw memory ?